<a href="https://colab.research.google.com/github/minjcop/machinelearning/blob/master/colabPreferences.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## 스펙 확인 및 구성

In [1]:
# OS 확인
!cat /etc/issue.net

Ubuntu 18.04.3 LTS


In [ ]:
# CPU
!cat /proc/cpuinfo

processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 79
model name	: Intel(R) Xeon(R) CPU @ 2.20GHz
stepping	: 0
microcode	: 0x1
cpu MHz		: 2200.000
cache size	: 56320 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 1
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 13
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx smap xsaveopt arat md_clear arch_capabilities
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs taa
bogomips	: 4400.00
clflush size	: 64
cache_alignment	: 64
address sizes	: 46 bits physical, 48 b

In [ ]:
# Memory
!cat /proc/meminfo

MemTotal:       13333556 kB
MemFree:        10523992 kB
MemAvailable:   12455016 kB
Buffers:           75936 kB
Cached:          2010800 kB
SwapCached:            0 kB
Active:           735836 kB
Inactive:        1787516 kB
Active(anon):     414320 kB
Inactive(anon):      340 kB
Active(file):     321516 kB
Inactive(file):  1787176 kB
Unevictable:           0 kB
Mlocked:               0 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Dirty:              1084 kB
Writeback:             0 kB
AnonPages:        436536 kB
Mapped:           222116 kB
Shmem:               964 kB
Slab:             166028 kB
SReclaimable:     126448 kB
SUnreclaim:        39580 kB
KernelStack:        3600 kB
PageTables:         5376 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     6666776 kB
Committed_AS:    2487144 kB
VmallocTotal:   34359738367 kB
VmallocUsed:           0 kB
VmallocChunk:          0 kB
Percpu:              928 kB
AnonHugePages:   

In [ ]:
# Disk
!df -h

Filesystem      Size  Used Avail Use% Mounted on
overlay          69G   32G   34G  49% /
tmpfs            64M     0   64M   0% /dev
tmpfs           6.4G     0  6.4G   0% /sys/fs/cgroup
shm             5.8G     0  5.8G   0% /dev/shm
tmpfs           6.4G   16K  6.4G   1% /var/colab
/dev/sda1        75G   33G   43G  44% /opt/bin
tmpfs           6.4G     0  6.4G   0% /proc/acpi
tmpfs           6.4G     0  6.4G   0% /proc/scsi
tmpfs           6.4G     0  6.4G   0% /sys/firmware


In [ ]:
# GPU, GPU 사용시에 확인 가능
!nvidia-smi

Mon Jul 20 07:40:23 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.51.05    Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla K80           Off  | 00000000:00:04.0 Off |                    0 |
| N/A   37C    P8    27W / 149W |      0MiB / 11441MiB |      0%      Default |
|                               |                      |                 ERR! |
+-------------------------------+----------------------+----------------------+
                                                                               
+-------

In [ ]:
# 구글 드라이브와 Colab 연동
from google.colab import auth
auth.authenticate_user()

from google.colab import drive
drive.mount('/content/gdrive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/gdrive


In [ ]:
# 특정 repository를 Colab에 연동
!git clone https://github.com/minjcop/machinelearning.git

Cloning into 'machinelearning'...
remote: Enumerating objects: 199, done.
remote: Counting objects: 100% (199/199), done.
remote: Compressing objects: 100% (156/156), done.
remote: Total 199 (delta 41), reused 198 (delta 40), pack-reused 0
Receiving objects: 100% (199/199), 12.65 MiB | 12.32 MiB/s, done.
Resolving deltas: 100% (41/41), done.


In [ ]:
ls -ltr

total 8
drwxr-xr-x 1 root root 4096 Jul 10 16:29 sample_data/
drwxr-xr-x 6 root root 4096 Jul 20 08:42 machinelearning/


In [ ]:
# tensorflow 설치(완료)
!pip install tensorflow==2.0.0-rc1

  Using cached https://files.pythonhosted.org/packages/40/d0/7359227ccfb6e2ae2a8fe345db1d513f010282022a7a6a899530cc63c41f/tensorflow-2.0.0rc1-cp36-cp36m-manylinux2010_x86_64.whl
  Found existing installation: tensorflow 2.2.0
    Uninstalling tensorflow-2.2.0:
      Successfully uninstalled tensorflow-2.2.0


In [ ]:
# Tensorflow 버전 확인, import tensorflow
from __future__ import absolute_import, division, print_function, unicode_literals
import tensorflow as tf
print(tf.__version__)

2.0.0-rc1


## tensorflow tutorial 예제 (출처 : https://www.tensorflow.org/tutorials/quickstart/beginner) 


In [ ]:
#mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

In [ ]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

In [ ]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 6s 105us/sample - loss: 0.3010 - accuracy: 0.9125
Epoch 2/5
60000/60000 [==============================] - 6s 95us/sample - loss: 0.1469 - accuracy: 0.9561
Epoch 3/5
60000/60000 [==============================] - 6s 94us/sample - loss: 0.1092 - accuracy: 0.9665
Epoch 4/5
60000/60000 [==============================] - 6s 97us/sample - loss: 0.0904 - accuracy: 0.9719
Epoch 5/5
60000/60000 [==============================] - 6s 102us/sample - loss: 0.0768 - accuracy: 0.9761
10000/1 - 1s - loss: 0.0409 - accuracy: 0.9747


[0.08047946909219027, 0.9747]

#### 훈련된 이미지 분류기가 약 97.5% 의 정확성을 보이는 것을 확인할 수 있다.